src/core.jl

```diff
@@ -76,10 +76,20 @@ end
 # all other functions should hit one of this eventually
 # Which handles the actual testing and user prompting
 
-function _test_reference(equiv, rendermode, file::File, actual::T) where T
+function test_reference(
+    file::File,
+    actual::T,
+    rendermode::Union{RenderMode, Nothing}=nothing,
+    equiv::Union{Function, Nothing}=nothing;
```

Comment:
Rather than defaulting to `nothing` how do you feel about defaulting to `default_rendermode(file, actual)` ?

Reply:
Cool, I thought default arguments are evaluated only *once* in compilation time.

Reply:
It turns out this doesn't work, I've added a comment for this in https://github.com/Evizero/ReferenceTests.jl/pull/37/commits/8ea330db3e1383877ca9c144fd6da784cd48952d